mod/01_rating/set/type/metric_value.rb

```diff
@@ -334,7 +332,24 @@ def special_editor? args
     end
   end
 
-  view :metric_company_add_value_editor do |_args|
+  def find_potential_sources company, metric
+    Card.search type_id: Card::SourceID,
+                right_plus: ['company', { refer_to: company }],
+                and: {
+                  right_plus: [
+                    'report_type',
+                    refer_to: {
+                      referred_to_by: metric + '+report_type' }] }
```

Comment:
in theory, this is supposed to be the canonical way to write this query:
````
Card.search(
  type_id: Card::SourceID,
  right_plus: [['company', { refer_to: company }],
               ['report_type', { refer_to: { referred_to_by: metric + '+report_type' } }]]
)
````
I can't say I'm 100% sure that it would work, but I wanted to let you know that there is supposed to be a way to do this without "and".